model/inline/src/main/scala/aqua/model/inline/ArrowInliner.scala

```diff
@@ -54,22 +54,34 @@ object ArrowInliner extends Logging {
     RawValueInliner
       .valueListToModel(results)
       .map(resolvedResults =>
-        // Fix the return values
-        (exportTo zip resolvedResults).map {
-          case (
+        (exportTo, resolvedResults) match {
+          // if export is a stream map and there is two results from function
+          case ((cexp @ CallModel.Export(n, st@StreamMapType(_))) :: Nil, (res1, desugar1) :: (res2, desugar2) :: Nil)  =>
+            (desugar1.toList ++ desugar2.toList :+ InsertKeyValueModel(res1, res2, n, st).leaf) -> (cexp.asVar :: Nil)
+          case _ =>
+            // Fix the return values
+            (exportTo zip resolvedResults).map {
+              case (
                 CallModel.Export(n, StreamType(_)),
                 (res @ VarModel(_, StreamType(_), _), resDesugar)
-              ) if !outsideStreamNames.contains(n) =>
-            resDesugar.toList -> res
-          case (
+                ) if !outsideStreamNames.contains(n) =>
+                resDesugar.toList -> res
+              case (
+                CallModel.Export(n, StreamMapType(_)),
+                (res @ VarModel(_, StreamMapType(_), _), resDesugar)
+                ) if !outsideStreamNames.contains(n) =>
```

Comment:
This guard does not help, as it is after the main case above, it should be before.
This code produces incorrect AIR:
```
func foo() -> %string, u64:
  map: %string
  <- map, 42

func main() -> string:
  m <- foo()
  <- m.get("key")!
```
AIR: `(ap (%map 42) %m)`